parlai/core/loader.py

```diff
@@ -381,7 +376,19 @@ def load_world_module(
             world_name = "SelfChatWorld"
         else:
             world_name = "DefaultWorld"
-    module_name = "%s.tasks.%s.worlds" % (repo, task)
+
+    if '.' in task_path_list[0]:
+        # The case of opt['task'] = 'parlai.tasks.squad.agents:DefaultTeacher'
+        # (i.e. specifying your own path directly)
+        module_name_parts = task_path_list[0].split('.')
+        if module_name_parts[-1] == 'agents':
+            # The world will be located in ".worlds", so correct the path
+            module_name = '.'.join(module_name_parts[:-1]) + '.worlds'
+        else:
+            module_name = task_path_list[0]
+    else:
+        task = task_path_list[0].lower()
+        module_name = "%s.tasks.%s.worlds" % (repo, task)
```

Comment:
Can we upgrade to f-strings while we're at it

Reply:
Yes, changed